src/OpenTelemetry/Metrics/CompositeMetricReader.cs

```diff
@@ -57,12 +60,117 @@ public CompositeMetricReader AddReader(MetricReader reader)
             };
             this.tail.Next = node;
             this.tail = node;
+            this.count++;
 
             return this;
         }
 
         public Enumerator GetEnumerator() => new Enumerator(this.head);
 
+        internal List<Metric> AddMetricsWithNoViews(Instrument instrument)
+        {
+            var metrics = new List<Metric>();
+            for (var cur = this.head; cur != null; cur = cur.Next)
```

Comment:
for a followup/todo - see if we need to catch exceptions from individual readers, so that anyone of them throwing won't affect the others.

Reply:
We don't expect this method to throw any exceptions, right?

Reply:
not sure. As a general principle, one reader's bad behaviour shouldn't affect other readers